pgml-extension/pgml_rust/src/backends/backend.rs

```diff
@@ -0,0 +1,42 @@
+use pgx::*;
+use serde::Deserialize;
+
+#[derive(PostgresEnum, Copy, Clone, Eq, PartialEq, Debug, Deserialize)]
+#[allow(non_camel_case_types)]
+pub enum Backend {
```

Comment:
Backend is a heavily overloaded term. I'd prefer something like `library`, `implementation`, `engine`, `provider`...